corehq/blobs/util.py

```diff
@@ -37,6 +39,79 @@ def pre_init(self, value, obj):
         return self.get_default() if value is None else value
 
 
+# extended from https://stackoverflow.com/a/31566082
+
+
+# extended from https://stackoverflow.com/a/31566082
+class GzipCompressReadStream:
+    CHUNK_SIZE = 4096
+
+    class Buffer:
+        def __init__(self):
+            self._buf = deque()
+            self._size = 0
+            self._content_length = None
+
+        @property
+        def content_length(self):
+            if self._content_length is None or self._size > 0:
+                raise Exception("content_length can't be accessed without completely reading the stream")
```

Comment:
This should use a custom exception subclass rather than `Exception`. As is, it is impossible to catch only this error.